### PR TITLE
APPDEV-1218 Profile pic update issue

### PR DESCRIPTION
**Pull request by madhuvardan-CMI**
- Fixed an issue where profile picture is not reflecting upon updating it.
- Refactored EditDetailsProfileFragment.java class

### CODE REVIEW

File: app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java [hunk 1]
```diff
@@ -417,6 +377,7 @@ private void redirectToNextPage()
 		{
 			showMobileVerificationScreen(null);
 		}
+		return null;
```
Comment (Bert-R) [inline app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java]: Add a comment like this:
```
return null; // Dummy return value to allow use as data load handler
```

File: app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java [hunk 2]
```diff
@@ -438,25 +400,19 @@ private void displayProfileImage(String path)
...
-					dialog.dismiss();
-				}
-			});
+			CustomAlertDialog.show(YonaActivity.getActivity(), getString(R.string.useralreadyregister), getString(R.string.ok), (dialog, which) -> dialog.dismiss());
 		}
 		else
 		{
 			Snackbar.make(YonaActivity.getActivity().findViewById(android.R.id.content), message.getMessage(), Snackbar.LENGTH_LONG).show();
 		}
+		return null;
```
Comment (Bert-R) [inline app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java]: Add a comment like this:
```
return null; // Dummy return value to allow use as data error handler
```

File: app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java [hunk 3]
```diff
@@ -283,79 +249,48 @@ private boolean userDetailsChanged()
...
 		}, AppConstant.TIMER_DELAY_HUNDRED);
-
 	}
 
 	private void profileEditMode()
 	{
 		profileImage.setOnClickListener(changeProfileImageClickListener);
 		updateProfileImage.setOnClickListener(changeProfileImageClickListener);
 		profileImageTxt.setBackground(ContextCompat.getDrawable(YonaActivity.getActivity(), R.drawable.bg_big_friend_round));
+		setupUserDetailsInUI();
```
[the remark this answers is not in the document]
Comment (madhuvardan-CMI) [reply app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java]: This isn't the fix. The fix is at lines (227-231 & 393)

File: app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java [hunk 4]
```diff
@@ -260,6 +221,11 @@ private void goToNext()
 			YonaAnalytics.createTapEventWithCategory(AnalyticsConstant.SCREEN_EDIT_PROFILE, AnalyticsConstant.SAVE);
 			updateUserProfile();
 		}
+		else if (isProfileUpdated)
+		{
+			YonaActivity.getActivity().onBackPressed();
+			YonaApplication.getEventChangeManager().notifyChange(EventChangeManager.EVENT_USER_UPDATE, YonaApplication.getEventChangeManager().getDataState().getUser());
```
Comment (madhuvardan-CMI) [inline app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java]: Here is the fix

Comment (Bert-R) [reply app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java]: I see. Thanks!

File: app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java [hunk 5]
```diff
@@ -369,45 +304,70 @@ else if (!APIManager.getInstance().getAuthenticateManager().isMobileNumberValid(
...
+		{
+			showErrorMessageToUserUponInvalidData(lastNameLayout, lastName, R.string.enternamevalidation);
+			return false;
+		}
+		else if (validateYonaFontEditTextView(nickName))
+		{
+			showErrorMessageToUserUponInvalidData(nickNameLayout, nickName, R.string.enternicknamevalidation);
+			return false;
+		}
+		else
```
[the remark this answers is not in the document]
Comment (Bert-R) [reply app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java]: You forgot this one.

File: app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java [hunk 6]
```diff
@@ -138,107 +144,65 @@ public void afterTextChanged(Editable s)
...
-		nickNameLayout = view.findViewById(R.id.nick_name_layout);
-		mobileNumberLayout = view.findViewById(R.id.mobile_number_layout);
-
+	private void setupFirstNameLayout(View view)
+	{
+		firstNameLayout = view.findViewById(R.id.first_name_layout);
 		firstName = view.findViewById(R.id.first_name);
 		firstName.addTextChangedListener(textWatcher);
-		firstName.setOnFocusChangeListener(onFocusChangeListener);
+		firstName.setOnFocusChangeListener(EditDetailsProfileFragment::onFocusChange);
```
Comment (Bert-R) [inline app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java]: By applying this pattern everywhere, you don't need the ``onFocusChangeListener`` member anymore.

**Pull request merged**